src/ngx_http_lua_kong_module.c

```diff
@@ -670,5 +683,197 @@ ngx_http_lua_kong_get_upstream_ssl_verify(ngx_http_request_t *r,
 
     return ctx->upstream_ssl_verify;
 }
+# endif
+
+char *
+ngx_http_lua_kong_load_var_index(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
+{
+
+    ngx_str_t                     *value;
+    ngx_int_t                      index;
+
+    value = cf->args->elts;
+
+    if (value[1].len == 0) {
+        ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+                           "invalid variable name size \"%V\"",
+                           &value[1]);
+        return NGX_CONF_ERROR;
+    }
+
+    index = ngx_http_get_variable_index(cf, &value[1]);
+
+    if (index == NGX_ERROR) {
+        ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+                           "undefined variable \"%V\"",
```

Comment:
I think `NGX_ERROR` returned basically means memory allocation failures, we should change this error message to be better reflecting that.
```suggestion
                           "unable to mark variable \"%V\" as indexed: no memory",
```